### PR TITLE
Instance name validation improvements

### PR DESCRIPTION
**Pull request by tomponline**
Based on suggestion from https://discuss.linuxcontainers.org/t/two-suggestions-to-improve-first-impression-with-lxd-3-19/6645 this PR expands the suggests to improve instance name validation error messages to make them more helpful.

Also validates snapshot names.

### CODE REVIEW

**Comment by tomponline** [comment]
Looking at issue with snapshot names validation.

File: lxd/instance/instance_utils.go [hunk 1]
```diff
@@ -872,3 +872,31 @@ func SuitableArchitectures(s *state.State, project string, req api.InstancesPost
...
+// so it takes an argument indicating whether the name is to be used for a snapshot or not.
+func ValidName(instanceName string, isSnapshot bool) error {
+	if isSnapshot {
+		parentName, snapshotName, _ := shared.InstanceGetParentAndSnapshotName(instanceName)
+		err := shared.ValidHostname(parentName)
+		if err != nil {
+			return errors.Wrap(err, "Invalid instance name")
+		}
+
+		err = shared.ValidHostname(snapshotName)
```
Comment (tomponline) [inline lxd/instance/instance_utils.go]: @stgraber this enforces the host name validation rules on the snapshot name part of a snapshot instance name. Is this OK (are there cases where a snapshot might have names that do not comply with those rules)?

Comment (stgraber) [reply lxd/instance/instance_utils.go]: Yeah, it's quite likely that some people have used dots for timestamps or names longer than 64 chars. There's no good reason for us to restrict snapshots the same way we restrict instances.

So a normal check of ensuring no slash or spaces should be enough.

Comment (tomponline) [reply lxd/instance/instance_utils.go]: @stgraber thanks thats fixed now.

**Pull request merged**